tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -355,8 +357,23 @@ export class WebGPUBackend extends KernelBackend {
     return this.currentComputePass;
   }
 
+  // Check if parallel compilation is done.
+  async checkCompileCompletion() {
+    const asyncProgramInfos = this.asyncProgramInfos;
+    this.asyncProgramInfos = [];
+    const pipelinesPromise = asyncProgramInfos.map(program => program.pipeline);
+    const pipelines = await Promise.all(pipelinesPromise);
+    for (let i = 0; i < pipelines.length; i++) {
+      this.pipelineCache[asyncProgramInfos[i].shaderKey] = pipelines[i];
+    }
+  }
+
   public async getBufferData(buffer: GPUBuffer, size: number):
       Promise<ArrayBuffer> {
+    if (env().getBool('ENGINE_COMPILE_ONLY')) {
+      console.warn('This can only be called when ENGINE_COMPILE_ONLY is false');
```

Comment:
How about changing the message to `The data may be invalid since ENGINE_COMPILE_ONLY is set` since it's a warning? And `return null` can be removed?

Reply:
We should tell user that data is invalid and how to avoid this. return null tells user that this data is invalid.